modules/tinymce/src/themes/silver/test/ts/headless/modes/UiReferencesTest.ts

```diff
@@ -0,0 +1,73 @@
+import { Gui, GuiFactory } from '@ephox/alloy';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Arr } from '@ephox/katamari';
+import { Class, Classes } from '@ephox/sugar';
+import { assert } from 'chai';
+
+import { LazyUiReferences, SinkAndMothership } from 'tinymce/themes/silver/modes/UiReferences';
+
+describe('headless.modes.UiReferencesTest', () => {
+  context('getUiMotherships', () => {
+
+    const makeSinkAndMothership = (className: string): SinkAndMothership => {
+      const mothership = Gui.create();
+      Class.add(mothership.element, className);
+      // It's not used as a sink in this test, so it doesn't need Positioning.
+      const sink = GuiFactory.build({
+        dom: {
+          tag: 'div'
+        }
+      });
+
+      return {
+        mothership,
+        sink
+      };
+    };
+
+    const assertClasses = (expected: string[][], motherships: Gui.GuiSystem[]) => {
+      const actual = Arr.map(motherships, (m) => Classes.get(m.element));
+      assert.deepEqual(actual, expected, 'Checking classes of motherships');
+    };
+
+    it('No UIs set', () => {
+      const lazyRefs = LazyUiReferences();
+      assert.deepEqual(lazyRefs.getUiMotherships(), [ ], 'There should be no motherships');
+    });
+
+    it('DialogUi set but not PopupUp', () => {
+      const lazyRefs = LazyUiReferences();
+      lazyRefs.dialogUi.set(
+        makeSinkAndMothership('alpha')
+      );
+      assertClasses([[ 'alpha' ]], lazyRefs.getUiMotherships());
+    });
+
+    it('PopupUi set but not DialogUi', () => {
+      const lazyRefs = LazyUiReferences();
+      lazyRefs.popupUi.set(
+        makeSinkAndMothership('beta')
+      );
+      assertClasses([[ 'beta' ]], lazyRefs.getUiMotherships());
+    });
+
+    it('PopupUi set and DialogUi set, but same component', () => {
```

Comment:
Should tests for this change have a jira ticket id prefix?

Reply:
Good point. I'll fix it in phase 3.

Reply:
It's hard to add them to all the tests changed here, because some things are just changes, but I'll add them to all new tests.